src/vcpkg/commands.portsdiff.cpp

```diff
@@ -39,18 +40,23 @@ namespace
         const auto checkout_this_dir =
             fmt::format("./{}", ports_dir_name); // Must be relative to the root of the repository
 
-        auto cmd = paths.git_cmd_builder(dot_git_dir, temp_checkout_path)
-                       .string_arg("checkout")
-                       .string_arg(git_commit_id)
-                       .string_arg("-f")
-                       .string_arg("-q")
-                       .string_arg("--")
-                       .string_arg(checkout_this_dir)
-                       .string_arg(".vcpkg-root");
-        cmd_execute_and_capture_output(cmd, default_working_directory, get_clean_environment());
-        cmd_execute_and_capture_output(paths.git_cmd_builder(dot_git_dir, temp_checkout_path).string_arg("reset"),
-                                       default_working_directory,
-                                       get_clean_environment());
+        RedirectedProcessLaunchSettings settings;
+        settings.environment = get_clean_environment();
+        flatten(cmd_execute_and_capture_output(paths.git_cmd_builder(dot_git_dir, temp_checkout_path)
+                                                   .string_arg("checkout")
+                                                   .string_arg(git_commit_id)
+                                                   .string_arg("-f")
+                                                   .string_arg("-q")
+                                                   .string_arg("--")
+                                                   .string_arg(checkout_this_dir)
+                                                   .string_arg(".vcpkg-root"),
+                                               settings),
+                Tools::GIT)
+            .value_or_exit(VCPKG_LINE_INFO);
```

Comment:
I believe this is a behavior change from previous -- if the git command failed, we would silently attempt to continue.

Agreed with the change, but highlighting it in case we see issues later.